### DISCUSSION TITLE
Adding configuration for rename pages directory

### DISCUSSION THREAD

**Discussion by Devisione** [vercel/next.js#30827] [Ideas]
### Describe the feature you'd like to request

while designing the project structure I faced the problem that the pages folder cannot be renamed.

In my view of the project architecture, I would like to see the "routes" folder which would contain a list of routes including the API, as well as the "pages" folder, which would contain a list of assets for pages.

### Describe the solution you'd like

add setting to next.config.js
{
     routesFolder: "pages" // by default
}

and the ability to redefine it in the future to any other.

### Describe alternatives you've considered

globally rename the pages folder and name it routes because apart from routing it cannot contain anything.

**Comment by epicbytes** [comment]
I am sure that this change will lead to misunderstanding, accumulate heterogeneous code of sites, and bring confusion to the development process, the recognition and transparency of the project code will be lost. I do not want to look for a folder that you decided to name anything but not the right one.

Excessive flexibility will ruin stability.

**Comment by Devisione** [comment]
in general, this should not affect recognition in any way, no one overrides such changes by default. Such a flexible configuration is needed in order to make it convenient to work with. There are quite a few configurations in next.config including even the "src" and "dist" directories.

answering your search question - everything will be stored in the configuration in one place as before. Such a change should untie the hands of the design of the project.
